docs/compute-engine/model_optimization_guide.md

```diff
@@ -0,0 +1,108 @@
+    sequence `QuantConv2D` → `ReLU` → `BatchNormalization`.
+
+    Note that a batch-norm can still be fused into a `QuantConv2D` layer without
+    a ReLU between them; however, a subsequent ReLU would not also be fused.
+
+*   **Avoid using Larq `QuantConv2D` layers with 'same-zero' padding**
+
+    The [Keras `Conv2D`
+    layer](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Conv2D)
+    supports either "valid" padding (no input padding) or "same" padding
+    (padding the input with zeroes so that the output has the same spatial
+    resolution). The Larq [`QuantConv2D`](/larq/api/layers/#quantconv2d) layer
+    also supports "valid" and "same" padding, but additionally has an attribute
+    `pad_values` that controls the padding constant that is used in the "same"
+    padding case. We refer to the combination of `padding="same"` with
+    `pad_values=0.0` (the default, to match the `Conv2D` layer) as 'same-zero'
+    padding, and the combination with `pad_values=1.0` as 'same-one' padding.
+
+    It is strongly recommended to use 'same-one' padding instead of 'same-zero'
+    padding; for an example of models using 'same-one' padding, see the
+    [QuickNet models in Larq
+    Zoo](https://github.com/larq/zoo/blob/bb0dbf7c13bcb21149cd19c3bd51fe7d885b3bd8/larq_zoo/sota/quicknet.py#L107-L108).
+
+    With 'same-zero' padding, because zero cannot be naturally represented in
+    binary data (which uses a single bit to represent either +1 or -1) a
+    zero-padding correction has to be added to values at the edge of the output
+    tensor in order to give the correct result. This extra computation can be
+    avoided by using 'same-one' padding.
+
+    Additionally, 'same-zero' padding prevents the converter op fusing
+    optimisations described above. These optimisations are not affected by
+    'same-one' padding.
+
+*   **Replace `QuantDense` layers with 1×1 `QuantConv2D`s**
+
+    LCE does not yet include an optimised binary dense op, which means that Larq
+    `QuantDense` layers are not accelerated in any way. The recommended
+    workaround for this is to replace all `QuantDense` layers with an equivalent
+    sequence of reshapes and a 1×1 `QuantConv2D` layer:
+
+    ```python
+    # Replace this...
+    x = lq.layers.QuantDense(num_dense_outputs, ...)(x)
+
+    # ...with this
+    x = tf.keras.layers.Reshape((1, 1, -1))(x)
+    x = lq.layers.QuantConv2D(num_dense_outputs, kernel_size=(1, 1), ...)(x)
+    x = tf.keras.layers.Reshape((-1,))(x)
+    ```
+
+*   **Ensure that `filter_height` × `filter_width` × `input_channels` is a
+    multiple of 128**
+
+    Inside the optimised binary convolution op, an `im2col` procedure is applied
+    to the input tensor, which reduces the convolution operation to a (binary)
+    matrix multiplication. The `im2col`-transformed input has a depth of size
+    `filter_height` × `filter_width` × `input_channels`, which is padded to a
```

Comment:
The same applies here, btw I think these detailed arguments explaining *why* these best practices are necessary are very nice :+1: